res/controllers/Roland_DJ-505-scripts.js

```diff
@@ -1538,8 +1538,8 @@ DJ505.PitchPlayMode = function (deck, offset) {
             this.outKey = "hotcue_" + this.number + "_enabled";
             this.output = function (value, group, control) {
                 var outval = this.outValueScale(value);
-                if (this.colorIdKey !== undefined && outval !== this.off) {
-                    this.outputColor(engine.getValue(this.group, this.colorIdKey));
+                if (this.colorKey !== undefined && outval !== this.off) {
+                    this.outputColor(engine.getValue(this.group, this.colorKey));
```

Comment:
I think it would be more convenient to use `color.colorFromHexCode` on the ControlObject value before passing it to `this.outputColor`.

Reply:
Maybe @Holzhaus can help ups here.
Can you have a look and check if I broke something?

Reply:
This does not work. `hotcue_X_color` contains an actual ARGB value, not the index of the color in the color palette.

Reply:
The HotcueButtons stopped working as well. Let me check what causes this.

Reply:
Ok, the problem is that the `colors` property of `HotcueButton` now needs to be an object with the different colors as keys. Here's a diff:

```diff
diff --git a/res/controllers/Roland_DJ-505-scripts.js b/res/controllers/Roland_DJ-505-scripts.js
index 2e2777c0e8..94dde87533 100644
--- a/res/controllers/Roland_DJ-505-scripts.js
+++ b/res/controllers/Roland_DJ-505-scripts.js
@@ -941,17 +941,17 @@ DJ505.PadColor = {
     DIM_MODIFIER: 0x10,
 };

-DJ505.PadColorMap = [
-    DJ505.PadColor.OFF,
-    DJ505.PadColor.RED,
-    DJ505.PadColor.GREEN,
-    DJ505.PadColor.BLUE,
-    DJ505.PadColor.YELLOW,
-    DJ505.PadColor.CELESTE,
-    DJ505.PadColor.PURPLE,
-    DJ505.PadColor.APRICOT,
-    DJ505.PadColor.WHITE,
-];
+DJ505.PadColorMap = {
+    0: DJ505.PadColor.OFF,
+    0xffc50a08: DJ505.PadColor.RED,
+    0xff32be44: DJ505.PadColor.GREEN,
+    0xff0044ff: DJ505.PadColor.BLUE,
+    0xfff8d200: DJ505.PadColor.YELLOW,
+    0xff42d4f4: DJ505.PadColor.CELESTE,
+    0xffaf00cc: DJ505.PadColor.PURPLE,
+    0xfffca6d7: DJ505.PadColor.APRICOT,
+    0xfff2f2fe: DJ505.PadColor.WHITE,
+}

 DJ505.PadSection = function (deck, offset) {
     // TODO: Add support for missing modes (flip, slicer, slicerloop)
@@ -1190,7 +1190,7 @@ DJ505.HotcueMode = function (deck, offset) {
     this.ledControl = DJ505.PadMode.HOTCUE;
     this.color = DJ505.PadColor.WHITE;

-    var hotcueColors = [this.color].concat(DJ505.PadColorMap.slice(1));
+    var hotcueColors = DJ505.PadColorMap;
     this.pads = new components.ComponentContainer();
     for (var i = 0; i <= 7; i++) {
         this.pads[i] = new components.HotcueButton({
@@ -1226,7 +1226,7 @@ DJ505.CueLoopMode = function (deck, offset) {
     this.ledControl = DJ505.PadMode.HOTCUE;
     this.color = DJ505.PadColor.BLUE;

-    var cueloopColors = [this.color].concat(DJ505.PadColorMap.slice(1));
+    var cueloopColors = DJ505.PadColorMap;
     this.PerformancePad = function(n) {
         this.midi = [0x94 + offset, 0x14 + n];
         this.number = n + 1;
@@ -1492,7 +1492,7 @@ DJ505.PitchPlayMode = function (deck, offset) {
     this.color = DJ505.PadColor.GREEN;
     this.cuepoint = 1;
     this.range = PitchPlayRange.MID;
-    var pitchplayColors = [this.color].concat(DJ505.PadColorMap.slice(1));
+    var pitchplayColors = DJ505.PadColorMap;

     this.PerformancePad = function(n) {
         this.midi = [0x94 + offset, 0x14 + n];
```